graphene/src/document.rs

```diff
@@ -119,13 +120,60 @@ impl Document {
 			.unwrap_or_default()
 	}
 
+	// Determines which layer is closer to the root
+	pub fn layer_closer_to_root(&self, A: &Vec<u64>, B: &Vec<u64>) -> bool {
+		let path_a = self.indices_for_path(A).unwrap();
+		let path_b = self.indices_for_path(B).unwrap();
+
+		let longest = max(path_a.len(), path_b.len());
+		for i in 0..longest {
+			// usize::MAX becomes negative one here, sneaky. So folders are compared as [X, -1].
+			let a = *path_a.get(i).unwrap_or(&usize::MAX) as i32;
+			let b = *path_b.get(i).unwrap_or(&usize::MAX) as i32;
+
+			// If these are equal it means we are comparing folder elements
+			if a == b {
+				continue;
+			}
+
+			// If this is smaller, it is closer to the root
+			if a < b {
+				return true;
+			}
```

Comment:
I am not really sure how this makes sense, if the layer id is a uuid, the ordering of ids doesn't really make sense here.
the proper way would probably be to find the ordering in the corresponding folder

Reply:
The id here is the indices, not a UUID. Check the above lines.

Reply:
I clarified the code / comments on this feedback, it would be confusing to anyone reading it.

Reply:
Thanks a lot, admittedly was pretty tired while reading the code but more comments are always welcome